### DISCUSSION TITLE
Maven-Multi-Module project and Quarkus-Extensions

### DISCUSSION THREAD

**Discussion by mschorsch** [quarkusio/quarkus#33148] [Q&A] [answered]
Say we have the following structure:
```
- pom.xml
  - server
    - pom.xml (contains quarkus-maven-plugin)
  - module A
    - pom.xml (contains a quarkus extension e.g. `quarkus-jaxb`)
  - module B
    - pom.xml
```
Is it necessary that module A is indexed by a jandex index or similar (https://quarkus.io/guides/cdi-reference#bean_discovery) so that the `quarkus-maven-plugin` takes module A into account? In the case of CDI beans this is necessary, but does this also apply to all other extensions (e.g. `quarkus-jaxb`)?

[a comment by an automated account is not shown]

**Comment by aloubyansky** [comment, marked as answer]
If the `server` depends on `module A` and `module A` depends on `quarkus-jaxb` then the `server` will have a transitive dependency on `quarkus-jaxb` which will be recognized as a Quarkus extension. There is no need to add any bean discovery config for that.

**Reply by mschorsch** [reply]
Thanks for the clarification
